app/src/main/java/com/hedvig/app/feature/marketing/ui/BackgroundImage.kt

```diff
@@ -0,0 +1,46 @@
+package com.hedvig.app.feature.marketing.ui
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.BoxScope
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.SideEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import coil.compose.rememberImagePainter
+import coil.size.Scale
+import com.google.accompanist.systemuicontroller.rememberSystemUiController
+import com.hedvig.app.feature.marketing.Background
+import com.hedvig.app.util.compose.rememberBlurHash
+
+@Composable
+fun BackgroundImage(background: Background, content: @Composable BoxScope.() -> Unit) {
+    Box(modifier = Modifier.fillMaxSize()) {
+        (background as? Background.Loaded)?.let { bg ->
```

Comment:
Could we maybe make this an early return if it's not Loaded instead of indenting everything below this? And let the compiler then infer that it's Background.Loaded since we're returning early if it's not.

Reply:
With the rewritten code, this is no longer possible 😄